### PR TITLE
ajax-y magic

### PR DESCRIPTION
**Pull request by jonbloom**
Theoretically this should allow the chat to be more chat-like and not require a page refresh.

### CODE REVIEW

**Comment by mreidsma** [comment]
Let's do a code review when I get back Tuesday! 

## 

Matthew Reidsma

I check email once a day, M-F.

File: index.php [hunk 1]
```diff
@@ -173,12 +172,13 @@
...
+                         <input type="hidden" name="userID" value="userID" id="userID"/>
 						<input type="submit" name="send message" value="Send Message" />
+
 				 </fieldset>
 			</form> 
-	    </div>  
-	
-	
+	    </div>
+    <script src="//code.jquery.com/jquery.js"></script>
```
Comment (mreidsma) [inline index.php]: Need to switch this to use the Google CDN, which is much faster and smaller.

File: messages.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?php
+session_start();
+error_reporting(E_ALL);
```
Comment (mreidsma) [inline messages.php]: No login checking - looks like I could load up messages.php and read and send messages by toggling the correct GET request.

File: js/scripts.js [hunk 3]
```diff
@@ -1,3 +1,43 @@
...
+function submitMessage(){
+    var message = jQuery("#message").val();
+    jQuery.ajax('messages.php', {
+        data: {addMessage: true,
+            message : message}
+    });
+    jQuery("#message").val('').focus();
+    return false;
+}
+jQuery(document).ready(function(e){
```
Comment (mreidsma) [inline js/scripts.js:39]: Smart way of doing this - looks like it has a nice php fallback. I like the ease with which you can change the refresh rate, too.

File: inc/functions.php [hunk 4]
```diff
@@ -44,7 +44,7 @@ function addMessage($user_id, $message) {
 	
 	$now = time();
 	
-	$add_message = $db->query("INSERT INTO transcript VALUES ('','$now','$user_id','$message')"); 
+	$add_message = $db->query("INSERT INTO transcript VALUES (NULL,'$now','$user_id','$message')");
```
Comment (mreidsma) [inline inc/functions.php:47]: Someday I will stop writing SQL queries like it's not 2003.

File: index.php [hunk 5]
```diff
@@ -57,6 +56,7 @@
 					$user_fname = $row["fname"];
 					$user_email = $row["email"];
 					$user_theme = $row["theme"];
+					$_SESSION['userid'] = $user_id;
```
Comment (mreidsma) [inline index.php:59]: Do we need the redundancy of this _and_ line 55? Why not update the rest of the $user_id call to pull from $_SESSION?

File: index.php [hunk 6]
```diff
@@ -29,7 +29,6 @@
 require 'inc/markdown.php';
 
 // Create database connection
-$db = new mysqli($db_host, $db_user, $db_pass, $db_database);
```
Comment (mreidsma) [inline index.php:32]: Why are you removing this db call? I can't see that $db is being defined anywhere now.

Comment (jonbloom) [inline index.php:32]: I added it to the config.php file. I probably should've put it in the sample as well. That way, it's available to the others such as messages.php.

File: inc/functions.php (hunk 4 above)
Comment (jonbloom) [inline inc/functions.php:47]: This was a challenging sentence to navigate.

File: index.php (hunk 6 above)
Comment (mreidsma) [inline index.php:32]: I'd prefer to leave the config file as just editable variables. If you want to move it out of index, let's put it in functions, where it makes logical sense.

Comment (mreidsma) [inline index.php:32]: In other news, this is great work! Once you put the connection in functions, I'll merge the request. A+.

Comment (jonbloom) [inline index.php:32]: All good to go!

**Pull request merged**